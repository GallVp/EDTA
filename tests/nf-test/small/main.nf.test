nextflow_pipeline {

    name "Test with a small genome"
    script "main.nf"

    test("small genome") {

        when {
            params {
                genome  = "https://raw.githubusercontent.com/jguhlin/EDTA/a2cd9a0777e4ac6e39545bacc3e752f94eb2f389/test/genome.fa"
                outdir  = "$outputDir"
            }
        }

        then {

            def stable_path = getAllFilesFromDir(
                params.outdir,
                false,
                [
                    'pipeline_info/*.{html,json,txt,yml}',
                    'raw/ltr/*.pp.LTR.intact.raw.gff3',
                    'raw/ltr/*.pp.LTR.raw.fa',
                    'raw/tir/*.pp.TIR.intact.raw.fa',
                    'raw/tir/*.pp.TIR.intact.raw.gff3',
                    'raw/line/*.pp.LINE.raw.fa',
                    'raw/line/*.pp.RM2.fa',
                    'raw/helitron/*.Helitron.intact.raw.fa',
                    'raw/helitron/*.Helitron.intact.raw.fa.anno.list',
                    'raw/helitron/*.Helitron.intact.raw.gff3'
                ],
                null,
                ['**']
            )

            def stable_name = getAllFilesFromDir(
                params.outdir,
                true,
                [
                    'pipeline_info/*.{html,json,txt,yml}'
                ],
                null,
                ['**']
            )

            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    [
                        'successful tasks': workflow.trace.succeeded().size(),
                        'versions': removeNextflowVersion("$outputDir/pipeline_info/software_versions.yml"),
                        'stable paths': stable_path,
                        'stable names': getRelativePath(stable_name, outputDir),
                    ]
                ).match() }
            )
        }

    }

}
