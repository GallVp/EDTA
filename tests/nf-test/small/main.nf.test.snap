{
    "small genome": {
        "content": [
            {
                "successful tasks": 25,
                "versions": {
                    "ANNOSINE": {
                        "annosine": "2.0.7"
                    },
                    "CAT_CAT": {
                        "pigz": "2.3.4"
                    },
                    "COMBINE_INTACT_TES": {
                        "perl": "v5.32.1"
                    },
                    "CUSTOM_SHORTENFASTAIDS": {
                        "python": "3.8.13",
                        "biopython": 1.75
                    },
                    "FINAL_FILTER": {
                        "perl": "v5.32.1",
                        "repeatmasker": "4.1.7-p1"
                    },
                    "FORMAT_HELITRONSCANNER_OUT": {
                        "perl": "v5.32.1"
                    },
                    "FORMAT_HELITRONSCANNER_OUT_EXT": {
                        "perl": "v5.32.1"
                    },
                    "HELITRONSCANNER_DRAW": {
                        "helitronscanner": "V1.1"
                    },
                    "HELITRONSCANNER_DRAW_RC": {
                        "helitronscanner": "V1.1"
                    },
                    "HELITRONSCANNER_POSTPROCESS": {
                        "perl": "v5.32.1",
                        "trf": 4.09,
                        "TEsorter": "1.4.7",
                        "mdust": "2006.10.17"
                    },
                    "HELITRONSCANNER_SCAN_HEAD": {
                        "helitronscanner": "V1.1"
                    },
                    "HELITRONSCANNER_SCAN_HEAD_RC": {
                        "helitronscanner": "V1.1"
                    },
                    "HELITRONSCANNER_SCAN_TAIL": {
                        "helitronscanner": "V1.1"
                    },
                    "HELITRONSCANNER_SCAN_TAIL_RC": {
                        "helitronscanner": "V1.1"
                    },
                    "LTRFINDER": {
                        "LTR_FINDER_parallel": "v1.1",
                        "ltr_finder": "v1.07"
                    },
                    "LTRHARVEST": {
                        "LTR_HARVEST_parallel": "v1.1",
                        "genometools": "1.6.5"
                    },
                    "LTRRETRIEVER_LTRRETRIEVER": {
                        "LTR_retriever": "v2.9.9"
                    },
                    "LTR_RETRIEVER_POSTPROCESS": {
                        "perl": "v5.32.1",
                        "trf": 4.09,
                        "TEsorter": "1.4.7",
                        "mdust": "2006.10.17"
                    },
                    "POST_LIBRARY_ANNOTATION": {
                        "perl": "v5.32.1",
                        "repeatmasker": "4.1.5"
                    },
                    "PROCESS_K": {
                        "perl": "v5.32.1",
                        "repeatmasker": "4.1.7-p1"
                    },
                    "REPEATMODELER_BUILDDATABASE": {
                        "repeatmodeler": "2.0.5"
                    },
                    "REPEATMODELER_POSTPROCESS": {
                        "perl": "v5.32.1",
                        "trf": 4.09,
                        "TEsorter": "1.4.7",
                        "mdust": "2006.10.17"
                    },
                    "REPEATMODELER_REPEATMODELER": {
                        "repeatmodeler": "2.0.5"
                    },
                    "TIRLEARNER": {
                        "TIR-Learner": "v3.0.2"
                    },
                    "TIR_LEARNER_POSTPROCESS": {
                        "perl": "v5.32.1",
                        "trf": 4.09,
                        "TEsorter": "1.4.7",
                        "mdust": "2006.10.17"
                    },
                    "Workflow": {
                        "oushujun/EDTA": "v3.0.0dev"
                    }
                },
                "stable paths": [
                    "genome.short.ids.tsv:md5,642382addc4beba37088b1ebe09d38cf"
                ],
                "stable names": [
                    "final",
                    "final/genome",
                    "final/genome/genome.TEanno.gff3",
                    "final/genome/genome.TElib.fa",
                    "final/genome/genome.intact.fa",
                    "final/genome/genome.intact.gff3",
                    "final/genome/genome.short.ids.tsv",
                    "pipeline_info"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-19T15:04:39.597839"
    }
}